### PR TITLE
hotfix-transfer-invalid-datas

### PR DESCRIPTION
**Pull request by Nexter73**

### CODE REVIEW

File: src/app/modules/timesheet/services/timesheet.service.ts [hunk 1]
```diff
@@ -37,17 +37,28 @@ export class TimesheetService {
   }
 
   public getTransferToken(): string {
```
Comment (RomainOliva) [inline src/app/modules/timesheet/services/timesheet.service.ts]: What do you mean about this:

```
    const transferedTimesheet =  this.getIfExistAlreadyPresentTimesheet(this.timesheet);

    if (this.timesheet.invoice) {
      transferedTimesheet.invoice: Object.assign({}, new Invoice(),  {
         provider: Object.assign(new Company(), this.timesheet.invoice.client),
         client: new Company()
      });
    }

    return this._serializer.serializeObject({
         mode: 'edit',
         timesheet: transferedTimesheet
    });
```

I don't test, I let you modify if needed :)

File: src/app/modules/timesheet/components/timesheet-review/timesheet-review.component.ts [hunk 2]
```diff
@@ -43,11 +43,18 @@ export class TimesheetReviewComponent implements OnInit {
         this.workingTime = this.calendarManager.getWorkedTime(this.timesheet);
 
         if (this.timesheet.invoice) {
+          console.log('IL y a une facture')
```
Comment (RomainOliva) [inline src/app/modules/timesheet/components/timesheet-review/timesheet-review.component.ts]: Don't forget to remove `console.log` ;)

File: src/app/modules/timesheet/services/timesheet.service.ts [hunk 3]
```diff
@@ -36,20 +36,68 @@ export class TimesheetService {
...
+
     return this._serializer.serializeObject({
-        mode: 'edit',
-        timesheet: transferedTimesheet
+      mode: 'edit',
+      timesheet: transferedTimesheet
     });
   }
 
+  // public getTransferToken(): string {
```
Comment (RomainOliva) [inline src/app/modules/timesheet/services/timesheet.service.ts]: same here, you can't remove this code ? :)

File: src/app/modules/timesheet/services/timesheet.service.ts [hunk 4]
```diff
@@ -36,20 +36,68 @@ export class TimesheetService {
...
```
[the remark this answers is not in the document]
Comment (Nexter73) [reply src/app/modules/timesheet/services/timesheet.service.ts]: It's others version of the method, I was waiting for your opinion on method before clean them :)

**Pull request merged**